plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -777,6 +777,20 @@ protected function prepare_object_for_database( $request, $creating = false ) {
 			}
 		}
 
+		$metadata_fields = array_filter(
+			$request->get_params(),
+			function( $k ) {
+				return str_starts_with( $k, 'meta_data_' );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+
+		if ( ! empty( $metadata_fields ) ) {
+			foreach ( $metadata_fields as $key => $value ) {
+				$product->update_meta_data( str_replace( 'meta_data_', '', $key ), $value );
```

Comment:
Should we still allow passing an object so consumers can take advantage of `id`?

https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php#L776